scripts/simplify.py

```diff
@@ -0,0 +1,215 @@
+
+def main(argv):
+    # Load and parse SVG
+	input_xml = minidom.parse(argv[0])
+	input_svg = input_xml.getElementsByTagName('svg')[0]
+
+	# Get 'layers' from SVG
+	layers = []
+
+	for element in input_xml.getElementsByTagName('g'):
+		# Sanity check
+		if element.getAttribute('inkscape:groupmode') != 'layer':
+			raise Exception('None layer groups are not supported at this time!')
+
+		if not element.getAttribute('pcbre:uuid'):
+			print('WARNING: No UUID assigned to layer. Skipping layer', element.getAttribute('inkscape:label'))
+			continue
+
+		layers += [element]
+
+	# Sort layers
+	layers.sort(key=lambda x: x.getAttribute('pcbre:uuid'))
+
+	# Create our 'clean' SVG object
+	clean_xml = minidom.Document()
+	clean_svg = clean_xml.createElement('svg')
+
+	# Recreate each allowed attribute
+	svg_allowed_attrs = ['width', 'height', 'viewBox']
+	for svg_allowed_attr in svg_allowed_attrs:
+		# Check if field has a value
+		if not input_svg.getAttribute(svg_allowed_attr):
+			continue
+	
+		clean_svg.setAttribute(svg_allowed_attr, input_svg.getAttribute(svg_allowed_attr))
+
+	# Add back XML namespcaes
+	input_svg_attributes = list(input_svg.attributes.keys()) if input_svg.attributes else []
+	for attribute in input_svg_attributes:
+		if attribute.startswith('xmlns:'):
+			clean_svg.setAttribute(attribute, input_svg.getAttribute(attribute))
+
+	# Sort attributes
+	sort_element_attr(clean_svg)
+
+	# Process each layer
+	layer_allowed_attrs = ['inkscape:label', 'transform', 'pcbre:uuid']
+	for layer in layers:
+		print('Processsing layer:', layer.getAttribute('inkscape:label'))
+
+		# Create group element
+		new_layer = clean_xml.createElement('g')
+
+		# Set default attributes
+		for layer_attr, value in default_layer_attrs.items():
+			new_layer.setAttribute(layer_attr, value)
+
+		# Recreate each allowed attribute in the newly created element
+		for svg_allowed_attr in layer_allowed_attrs:
+			# Check if field has a value
+			if not layer.getAttribute(svg_allowed_attr):
+				continue
+		
+			new_layer.setAttribute(svg_allowed_attr, layer.getAttribute(svg_allowed_attr))
+
+		# Add UUID if needed
+		if not element.getAttribute('pcbre:uuid'):
+			element.setAttribute('pcbre:uuid', str(uuid.uuid4()))
+
+		# Sort attributes
+		sort_element_attr(new_layer)
+
+		# Sub elements of current layer
+		if layer.getAttribute('inkscape:label') in ['Board Bottom', 'Board Top']:
+			process_image_layer(clean_xml, layer, new_layer)
+		elif layer.getAttribute('inkscape:label') in ['Bottom Zones', 'Top Zones']:
+			process_zones_layer(clean_xml, layer, new_layer)
+		elif layer.getAttribute('inkscape:label') in ['Top Traces', 'Bottom Traces']:
+			process_traces_layer(clean_xml, layer, new_layer)
+
+		#
+		clean_svg.appendChild(new_layer)
+
+	# Append SVG object to the document
+	clean_xml.appendChild(clean_svg)
+
+	# Write out simplified SVG
+	open('./clean.svg', 'wb').write(clean_xml.toprettyxml('\t', '\n', 'UTF-8'))
```

Comment:
Filename should probably be an argument.
We then probably want some script that we *could* run from a git hook or CI, so all SVGs can automatically be simplified.

Reply:
Fixed. 1c7ceb1fbbade2d6de2ad3e17e6a3849d2b55e63